components/constellation/constellation.rs

```diff
@@ -2474,21 +2464,19 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
         let event = MozBrowserEvent::Error(MozBrowserErrorType::Fatal, reason, report);
 
-        if let Some(pipeline_id) = pipeline_id {
-            if let Some((ancestor_id, mozbrowser_iframe_id)) = self.get_mozbrowser_ancestor_info(pipeline_id) {
-                if let Some(ancestor) = self.pipelines.get(&ancestor_id) {
-                    return ancestor.trigger_mozbrowser_event(Some(mozbrowser_iframe_id), event);
-                }
-            }
-        }
-
-        if let Some(root_frame) = self.frames.get(&self.root_frame_id) {
-            if let Some(root_pipeline) = self.pipelines.get(&root_frame.current.pipeline_id) {
-                return root_pipeline.trigger_mozbrowser_event(None, event);
-            }
-        }
-
-        warn!("Mozbrowser error after root pipeline closed.");
+        match self.frames.get(&top_level_frame_id) {
+            None => warn!("Mozbrowser error after top-level frame closed."),
+            Some(frame) => match self.pipelines.get(&frame.current.pipeline_id) {
```

Comment:
Re-reading that code. I was confused about using `frame.current`, which is only used to get the parent frame (not to identify the failing pipeline).